### PR TITLE
SpinButton accessibility labelling

### PR DESCRIPTION
**Pull request by AntonioGugin**
Adding Accessibility ESlint rule for SpinButton
[Issue](https://github.com/microsoft/eslint-plugin-fluentui-jsx-a11y/issues/50)

### CODE REVIEW

File: tests/lib/rules/spin-button-needs-labelling-v9.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+ruleTester.run("spin-button-needs-labelling-v9", rule, {
+    valid: [
+        "<><Label>This is a SpinButton<SpinButton checked={true} /></Label></>",
+        `<SpinButton label="This is a spin button" checked={true} />`,
+        `<SpinButton label="This is a spin button" checked={true}></SpinButton>`,
```
[the remark this answers is not in the document]
Comment (AntonioGugin) [reply tests/lib/rules/spin-button-needs-labelling-v9.js]: Not supported apparently

File: lib/rules/spin-button-needs-labelling-v9.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                if (elementType(node) !== "SpinButton") {
+                    return;
+                }
+
+                // if the SpinButton has a label, if the SpinButton has an associated label, return
+                if (
+                    hasNonEmptyProp(node.attributes, "label") ||
+                    isInsideLabelTag(context) ||
+                    hasAssociatedLabelViaHtmlFor(node, context) ||
+                    hasAssociatedLabelViaAriaLabelledBy(node, context)
```
Comment (aubreyquinn) [inline lib/rules/spin-button-needs-labelling-v9.js:45]: Today I found that it also works if wrapped in a Label component. I think there is a utility function to check for that.

 ```
     <Label>
        Wrapped SpinButton
        <SpinButton defaultValue={10} min={0} max={20} id={id} />
      </Label>
```

File: lib/rules/spin-button-needs-labelling-v9.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        return {
+            // visitor functions for different types of nodes
+            JSXOpeningElement(node) {
+                // if it is not a SpinButton, return
+                if (elementType(node) !== "SpinButton") {
+                    return;
+                }
+
+                // if the SpinButton has a label, if the SpinButton has an associated label, return
+                if (
```
[the remark this answers is not in the document]
Comment (aubreyquinn) [reply lib/rules/spin-button-needs-labelling-v9.js:41]: I found that it does work but it's not recommended

File: lib/rules/spin-button-needs-labelling-v9.js (hunk 2 above)
Comment (aubreyquinn) [reply lib/rules/spin-button-needs-labelling-v9.js:45]: Never mind, we got this already :)

File: lib/rules/spin-button-needs-labelling-v9.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                    hasNonEmptyProp(node.attributes, "label") ||
```
Comment (aubreyquinn) [inline lib/rules/spin-button-needs-labelling-v9.js:42]: I don't think that the SpinButton has a label prop actually. 

Text content - component does not have	
aria-label	yes - works but not recommended	
htmlFor - yes	[done]
aria-labelledby- yes [done]	
wrapped in tool tip - yes - works but not recommended	
wrapped in label- yes	[done]
label prop - component does not have	
content prop - component does not have

File: docs/rules/spin-button-unrecommended-labelling-v9.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+Examples of **unrecommended** code for this rule:
+
+```jsx
+<SpinButton defaultValue={10} min={0} max={20} aria-label={"my-aria-label-1"} />
+```
+
+```jsx
+
+<Tooltip content="Sping Button label" relationship="label">
```
Comment (nikitasingla-MSFT) [inline docs/rules/spin-button-unrecommended-labelling-v9.md:26]: Can we remove this from the readme as tooltip is not recommended for spin button, we should not recommend this through the readme in my pov, @aubreyquinn what are your thoughts on this?

we should have the rule, but lets avoid it in readme.

Comment (AntonioGugin) [reply docs/rules/spin-button-unrecommended-labelling-v9.md:26]: This talks about unrecommended ways of doing this. Same as other readmes talk about incorrect ones

Comment (aubreyquinn) [reply docs/rules/spin-button-unrecommended-labelling-v9.md]: Yes, I agree :)

**Review by nikitasingla-MSFT — APPROVED** [review]

**Pull request merged**